pkg/roachprod/vm/gce/gcloud.go

```diff
@@ -1649,6 +1649,45 @@ func computeGrowDistribution(groups []jsonManagedInstanceGroup, newNodeCount int
 	return addCount
 }
 
+// Shrink shrinks the cluster by deleting the given VMs. This is only supported
+// for managed instance groups. Currently, nodes should only be deleted from the
+// tail of the cluster, due to complexities thar arise when the node names are
+// not contiguous.
+func (p *Provider) Shrink(l *logger.Logger, vmsToDelete vm.List, clusterName string) error {
+	if !isManaged(vmsToDelete) {
+		return errors.New("shrinking is only supported for managed instance groups")
+	}
+
+	project := vmsToDelete[0].Project
```

Comment:
Do we have to worry about `vmsToDelete[i].Project != vmsToDelete[0].Project`?

Reply:
Technically it shouldn't be possible to have a cluster that has been created over multiple projects, we also check earlier for grow/shrink that only one provider is involved.